src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -451,34 +441,33 @@ internal static bool CanHaveExecutableCodeBlock(ISymbol symbol)
 
         internal void ExecuteAndCatchIfThrows(DiagnosticAnalyzer analyzer, Action analyze)
         {
-            ExecuteAndCatchIfThrows(analyzer, analyze, _addExceptionDiagnostic, _continueOnAnalyzerException, _cancellationToken);
+            ExecuteAndCatchIfThrows(analyzer, analyze, _onAnalyzerException, _cancellationToken);
         }
 
         private static void ExecuteAndCatchIfThrows(
```

Comment:
this seems really generic helper. 

ExceptionHelper.ExecuteAndCatchIfThrows(Action action, Action<Exception> onException, CancellationToken c)

since analyze already capture, there seems no advantage saving one allocation for onException